terraform/modules/kubernetes/kubernetes.tf

```diff
@@ -205,15 +201,48 @@ resource "kubernetes_cron_job" "workflow_manager" {
             # "Never".
             # https://kubernetes.io/docs/concepts/workloads/controllers/job/#handling-pod-and-container-failures
             # https://github.com/kubernetes/kubernetes/issues/74848
-            restart_policy       = "Never"
-            service_account_name = kubernetes_service_account.workflow_manager.metadata[0].name
+            restart_policy                  = "Never"
+            service_account_name            = kubernetes_service_account.workflow_manager.metadata[0].name
+            automount_service_account_token = true
           }
         }
       }
     }
   }
 }
 
+resource "kubernetes_role" "workflow-manager-role" {
+  metadata {
+    name      = "${var.environment}-${var.data_share_processor_name}-wfm-role"
+    namespace = var.kubernetes_namespace
+  }
+
+  rule {
+    api_groups = [""]
```

Comment:
I find it surprising that an empty string here is implicitly the "core" API group, so that's probably worth a comment, we well as one explaining what privileges we're granting this role and why.